planemo/shed/__init__.py

```diff
@@ -261,7 +261,7 @@ def upload_repository(ctx, realized_repository, **kwds):
         return report_non_existent_repository(realized_repository)
 
     if kwds.get("check_diff", False):
-        is_diff = diff_repo(ctx, realized_repository, **kwds)
+        is_diff = diff_repo(ctx, realized_repository, **kwds) != 0
```

Comment:
Wait - this doesn't actually change the behavior - not sure what I was thinking :angry:.